vcn_gateways.tf

```diff
@@ -79,18 +132,64 @@ resource "oci_core_route_table" "nat" {
   freeform_tags = var.tags
 
   route_rules {
+    # * With this route table, NAT Gateway is always declared as the default gateway
     destination       = local.anywhere
     destination_type  = "CIDR_BLOCK"
     network_entity_id = oci_core_nat_gateway.nat_gateway[0].id
+    description       = "Terraformed - Auto-generated at NAT Gateway creation: NAT Gateway as default gateway"
   }
 
   dynamic "route_rules" {
+    # * If Service Gateway is created with the module, automatically creates a rule to handle traffic for "all services" through Service Gateway
     for_each = var.service_gateway_enabled == true ? list(1) : []
 
     content {
       destination       = lookup(data.oci_core_services.all_oci_services[0].services[0], "cidr_block")
       destination_type  = "SERVICE_CIDR_BLOCK"
       network_entity_id = oci_core_service_gateway.service_gateway[0].id
+      description       = "Terraformed - Auto-generated at Service Gateway creation: All Services in region to Service Gateway"
+    }
+  }
+
+  dynamic "route_rules" {
```

Comment:
you also need to add a condition to add this rule if the DRG is created

Reply:
Same as above.

This block will be evaluated only if the parent resource (route table) is created, which itself have a test to be created based on var.service_gateway_enabled value.

I don't think we have to check var.service_gateway_enabled value again for dynamic block creation.

Reply:
this is for nat gw. what happens when you create the nat gw but not the drg?

Reply:
same answer as above